docs/sdk/technical-reference/smart-order-router.md

```diff
@@ -2,19 +2,64 @@
 order: 1
 ---
 # Smart Order Router
+The Smart Order Router (SOR) is a tool designed to optimize order routing across Balancer pools for the best possible price execution. It can be accessed through the SDK, but it is also available as a standalone package for those who only need the swap routing functionality. This example code will utilize the SDK, but it is easy to understand how to make changes for the standalone SOR package.
 
-The Smart Order Router (SOR) is a library for order routing optimzation across Balancer pools for best price execution. It is a component of the SDK but also exists as a standalone package for integrators who only need swap routing logic. The example code and reference shown in the docs will use the SDK, but it should be straightforward to infer the changes required for the standalone SOR package.
+The below docs assume the SDK is installed and initialized.
 
-[![npm version](https://img.shields.io/npm/v/@balancer-labs/sor/latest.svg)](https://www.npmjs.com/package/@balancer-labs/sor/v/latest)
+```javascript
+import { BalancerSDK } from '@balancer-labs/sdk'
 
-::: details Preconfig
-The below docs assume the SDK is installed and initialized. See [Intro](../overview/README.md) for instructions
-:::
+const balancer = new BalancerSDK({
+  network: 1, // Mainnet
+  rpcUrl: 'https://rpc.ankr.com/eth' // rpc endpoint
+})
 
-## SOR Basics
+const { swaps } = balancer // Swaps module is abstracting SOR
+```
 
-TODO: explain the general flow of a SOR request, including pool data fetching, route proposal, route optimization, return data format, and what's required to convert that to a tx. And details around the decision points a developer needs to make for things like `useBpts`, if the relayer is required, etc.
+The general flow for finding a trade route using SOR (Smart Order Router) includes the following steps:
 
-```typescript
+### Step 1. Pool data fetching
+The SOR requires information about the available pools and their current status, including the prices of tokens and the liquidity of the pools. It is essential to use the SOR based on up-to-date information, as outdated information can lead to incorrect slippage predictions and potentially result in failed swaps.
+```javascript
+await swaps.fetchPools()
+```
 
-```
\ No newline at end of file
+### Step 2. Route proposal
+The SOR determines the optimal trade route based on the available pool data and the desired trade, taking into account factors such as trade size, gas costs, and slippage. When searching for swaps, developers have to choose between two types of swaps:
+
+* `findRouteGivenIn`, where the amount of tokens being sent to the pool is known, or
+* `findRouteGivenOut`, where the amount of tokens received from the pool is known.
+
+```javascript
+const swapInfo = await swaps.findRouteGivenIn({
```

Comment:
It's SDK abstracted SOR vs naked SOR. I agree it will be too confusing, so I updated everything to use SDK swaps module with findRouteGivenIn / findRouteGivenOut. In join / exits we use ExactIn / ExactOut - that naming comes from smart contracts. I think we should be consistent and use the same naming here, eg: findRouteExactOut.